js/compiler_test.ts

```diff
@@ -88,27 +92,26 @@ const modBModuleInfo = mockModuleInfo(
   "/root/project/modB.ts",
   MediaType.TypeScript,
   modBSource,
+  undefined,
   undefined
 );
 
-// TODO(#23) Remove source map strings from fooBarTsOutput.
 // tslint:disable:max-line-length
 const fooBarTsOutput = `define(["require", "exports", "deno"], function (require, exports, deno) {
     "use strict";
     Object.defineProperty(exports, "__esModule", { value: true });
     console.log(deno);
     exports.foo = "bar";
 });
-//# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoiYmFyLmpzIiwic291cmNlUm9vdCI6IiIsInNvdXJjZXMiOlsiZmlsZTovLy9yb290L3Byb2plY3QvZm9vL2Jhci50cyJdLCJuYW1lcyI6W10sIm1hcHBpbmdzIjoiOzs7SUFDQSxPQUFPLENBQUMsR0FBRyxDQUFDLElBQUksQ0FBQyxDQUFDO0lBQ0wsUUFBQSxHQUFHLEdBQUcsS0FBSyxDQUFDIiwic291cmNlc0NvbnRlbnQiOlsiaW1wb3J0ICogYXMgZGVubyBmcm9tIFwiZGVub1wiO1xuY29uc29sZS5sb2coZGVubyk7XG5leHBvcnQgY29uc3QgZm9vID0gXCJiYXJcIjtcbiJdfQ==
+//# sourceMappingURL=bar.js.map
```

Comment:
I don't know if this matters, but I found it a little strange that when I catted one of the generated JS files, it has 
`sourceMappingURL=print_hello.js.map` but that file doesn't exist anywhere, instead the corresponding source map file is `"37d9f09dbaa994246e3bc83a56d9d87810008637.js.map"`
```
> cat ./gen/37d9f09dbaa994246e3bc83a56d9d87810008637.js
define(["require", "exports"], function (require, exports) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    function printHello() {
        console.log("Hello");
    }
    exports.printHello = printHello;
});
//# sourceMappingURL=print_hello.js.map
//# sourceURL=/Users/rld/src/deno/tests/subdir/print_hello.ts
```

Reply:
So that comes from TypeScript, and it doesn't know the generated file names.  The compiler doesn't know the generated file name either, because if it is coming from the cache, it wouldn't usually be re-transpiling.  We would have to do a fair bit of work to make that valid, and even then, it wouldn't help in the way we actually resolve the map anyways.  The compiler has no clue, as is appropriate, that there is a cache of modules, and leaking that abstraction further would be bad IMO.

This means that the source map wouldn't work out of the compiler context.  You could argue that any imports into a module are also named "incorrectly" and wouldn't work on their own, without running inside a Deno context.